### PR TITLE
ARROW-5265: [Python][CI] Add integration test with kartothek

### PR DESCRIPTION
**Pull request by xhochy**
Kartothek broke again with the latest Arrow master branch (e.g. https://issues.apache.org/jira/browse/ARROW-8057) thus I finally took the time and wrote the integration test. This will be failing until we have a workaround implement in kartothek for ARROW-8057.

cc @fjetter

### CODE REVIEW

**Comment by wesm** [comment]
Note that ARROW-8057 has not yet been released.

**Comment by wesm** [comment]
The underlying issue should be fixed in https://github.com/apache/arrow/pull/6569

**Comment by wesm** [comment]
@kszucs @nealrichardson is there any mailing list-level visibility of GitHub Actions cron jobs? What happens when these fail?

File: docker-compose.yml [hunk 1]
```diff
@@ -666,6 +666,36 @@ services:
...
+    # Possible $TURBODBC parameters:
+    #  - `latest`: latest release
+    #  - `master`: git master branch, use `docker-compose run --no-cache`
+    #  - `<version>`: specific version available under github releases
+    # Usage:
+    #   docker-compose build conda-cpp
+    #   docker-compose build conda-python
+    #   docker-compose build conda-python-kartothek
+    #   docker-compose run --rm conda-python-kartothek
+    image: ${REPO}:${ARCH}-conda-python-${PYTHON}-kartothek-${TURBODBC}
```
Comment (fjetter) [inline docker-compose.yml]: what exactly is the turbodbc coupling here?

**Review by fjetter — COMMENTED** [review]
I'm actually impressed how easy this integration setup is 👏

File: docker-compose.yml (hunk 1 above)
Comment (xhochy) [reply docker-compose.yml]: Copy-paste failure by me.

**Comment by kszucs** [comment]
We don't have mailing list-level visibility of GitHub Actions cron jobs, when these the watchers or members (I'm not sure, depending on individuals' notification settings) get emails with subject like:

```
[apache/arrow] Run failed: Python Cron - master (fe71388)
```

File: Makefile.docker [hunk 2]
```diff
@@ -29,7 +29,7 @@ DEPENDS_ON_CPP = build-c_glib build-python build-r
 DEPENDS_ON_CPP_ALPINE = build-python-alpine
 DEPENDS_ON_PYTHON = build-lint build-docs build-dask build-hdfs-integration \
                     build-spark-integration build-python-nopandas \
-                    build-turbodbc-integration
+                    build-turbodbc-integration build-kartothek-integration
```
Comment (kszucs) [inline Makefile.docker]: @xhochy we should remove this makefile, this is outdated and we should rather have a tool in achery.

File: ci/docker/conda-python-kartothek.dockerfile [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        storefact \
+        toolz \
+        urlquote \
+        zstandard && \
+    conda clean --all
+
+ARG kartothek=latest
+COPY ci/scripts/install_kartothek.sh /arrow/ci/scripts/
+RUN /arrow/ci/scripts/install_kartothek.sh ${kartothek} /kartothek
+
```
Comment (kszucs) [inline ci/docker/conda-python-kartothek.dockerfile]: Newline.

**Comment by nealrichardson** [comment]
We could make the crossbow email job also query the GHA nightly build status. I hadn't seen that they'd added an API for that, which would be nice, but even without a proper API, we could find a way. One lame workaround that occurred to me was that we could fetch the public badge SVG and grep for "failure".

**Comment by wesm** [comment]
If you rebase hopefully you can get a passing build. Let me know

**Comment by kszucs** [comment]
> We could make the crossbow email job also query the GHA nightly build status. I hadn't seen that they'd added an API for that, which would be nice, but even without a proper API, we could find a way. One lame workaround that occurred to me was that we could fetch the public badge SVG and grep for "failure".

Github actions already has an API which we can use.

**Comment by kszucs** [comment]
@xhochy @fjetter seems like the Kartothek build is failing

**Comment by fjetter** [comment]
@kszucs on it. https://github.com/JDASoftwareGroup/kartothek/pull/240 should fix it (waiting for CI to finish)

**Comment by kszucs** [comment]
@fjetter this won't resolve the `latest` test which tests against the latest kartothek release, but will probably solve the `master` test. I can switch to run only the `master` test until the next release of kartothek comes out, then we can keep compatibility with both the latest release and the master. Does is sound good?

**Comment by fjetter** [comment]
I'm fine with either configuration. I intend to release another Kartothek version later this day. I can also change this to latest once the Conda package is available.

The builds are still running against latest. Is there something wrong with the script?

File: Makefile.docker (hunk 2 above)
Comment (xhochy) [reply Makefile.docker]: Feel free to remove it. I just wanted to keep everything here uptodate.

**Comment by kszucs** [comment]
The build failures are unrelated, merging. Thanks @fjetter for the quick fixes!

**Review by kszucs — APPROVED** [review]

**Pull request closed without merging**